x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_indices/search_indices.tsx

```diff
@@ -5,152 +5,65 @@
 
   const [calloutDismissed, setCalloutDismissed] = useLocalStorage<boolean>(
     'enterprise-search-indices-callout-dismissed',
     false
   );
 
-  useEffect(() => {
-    makeRequest({ meta });
-  }, [meta.page.current]);
-
-  const columns = [
-    {
-      field: 'name',
-      name: i18n.translate('xpack.enterpriseSearch.content.searchIndices.name.columnTitle', {
-        defaultMessage: 'Index name',
-      }),
-      render: (name: string) => (
-        <EuiLinkTo
-          data-test-subj="search-index-link"
-          to={generatePath(SEARCH_INDEX_PATH, { indexName: name })}
-        >
-          {name}
-        </EuiLinkTo>
-      ),
-      sortable: true,
-      truncateText: true,
-    },
-    {
-      field: 'total.docs.count',
-      name: i18n.translate('xpack.enterpriseSearch.content.searchIndices.docsCount.columnTitle', {
-        defaultMessage: 'Docs count',
-      }),
-      sortable: true,
-      truncateText: true,
-    },
-    {
-      field: 'health',
-      name: i18n.translate('xpack.enterpriseSearch.content.searchIndices.health.columnTitle', {
-        defaultMessage: 'Index health',
-      }),
-      render: (health: 'red' | 'green' | 'yellow' | 'unavailable') => (
-        <span>
-          <EuiIcon type="dot" color={healthColorsMap[health] ?? ''} />
-          &nbsp;{health ?? '-'}
-        </span>
-      ),
-      sortable: true,
-      truncateText: true,
-    },
-    {
-      field: 'data_ingestion',
-      name: i18n.translate(
-        'xpack.enterpriseSearch.content.searchIndices.dataIngestion.columnTitle',
-        {
-          defaultMessage: 'Data ingestion',
-        }
-      ),
-      render: (dataIngestionStatus: string) =>
-        dataIngestionStatus ? (
-          <EuiBadge color={dataIngestionStatus === 'connected' ? 'success' : 'warning'}>
-            {dataIngestionStatus}
-          </EuiBadge>
-        ) : null,
-      truncateText: true,
-    },
-    {
-      align: 'right' as HorizontalAlignment,
-      field: 'total.store.size_in_bytes',
-      name: i18n.translate('xpack.enterpriseSearch.content.searchIndices.storage.columnTitle', {
-        defaultMessage: 'Storage',
-      }),
-      sortable: true,
-      truncateText: true,
-    },
-    {
-      actions: [
-        {
-          render: ({ name }: SearchIndex) => (
-            <EuiButtonIconTo
-              aria-label={name}
-              iconType="eye"
-              data-test-subj="view-search-index-button"
-              to={generatePath(SEARCH_INDEX_PATH, {
-                indexName: name,
-              })}
-            />
-          ),
-        },
-      ],
-      name: i18n.translate('xpack.enterpriseSearch.content.searchIndices.actions.columnTitle', {
-        defaultMessage: 'Actions',
-      }),
-    },
-  ];
+  useDebounce(
+    () => makeRequest({ meta, returnHiddenIndices: showHiddenIndices, searchQuery }),
+    150,
+    [searchQuery, meta.page.current, showHiddenIndices]
+  );
```

Comment:
Can we use [kea breakpoints](https://v2.keajs.org/docs/guide/additional#breakpoints) for this?

Reply:
Ooh, I'd forgotten about that feature. Yes, definitely, much better location for it!